src/charm.py

```diff
@@ -233,6 +235,45 @@ def _on_restart_action(self, event):
         container.restart(self.name)
         self.set_active_unit_status()
 
+    @log_event_handler(logger)
+    def _on_update_status(self, event):
+        """Handle `update-status` events.
+
+        Args:
+            event: The `update-status` event triggered at intervals.
+        """
+        try:
+            self._validate()
+        except ValueError:
+            return
+
+        container = self.unit.get_container(self.name)
+        valid_pebble_plan = self._validate_pebble_plan(container)
+        if not valid_pebble_plan:
+            self._update(event)
```

Comment:
So, if it's not valid you push an update? Is this correct?

Reply:
I think theoretically, we could just return here without pushing the update, because if the pebble plan is missing for some reason then another hook should be coming up like `temporal_pebble_ready` as you say. But I don't know enough to reason whether that is always true or not.

Reply:
Yes, if the pebble plan is invalid, we push an update, which attempts to create a new pebble layer. I don't believe there's too much difference here other than the fact that we are not waiting for the `temporal_pebble_ready` event. Either way, we are calling the `_update(event)` method, but in this case we are calling it without waiting for the `temporal_pebble_ready` event.

Reply:
I agree that theoretically we could just return here and wait for the `pebble_ready`. I did see an edge case with the Superset charm where the `update_status` was called before the `pebble_ready` in which case this would create the pebble plan more quickly by calling it directly. 

It seems that the `pebble_ready` event fires if the `container.can_connect()` is successful. As you are also checking for this in your `_validate` step I don't foresee and issue with calling `_update` directly without the `temporal_pebble_ready`. But I don't think it is strictly necessary.